tools/download_file.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+# Copyright (c) 2012 The Chromium Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+from __future__ import print_function
+import argparse
+import os
+import sys
+import time
+
+try:
+    from urllib2 import HTTPError, URLError, urlopen
+except ImportError:  # For Py3 compatibility
+    from urllib.error import HTTPError, URLError
+    from urllib.request import urlopen
+
+SECONDS_PER_PROGRESS = 5
+
+def DownloadUrl(url, output_file):
+    """Download url into output_file."""
+    CHUNK_SIZE = 4096
+    TOTAL_DOTS = 10
+    num_retries = 3
+    retry_wait_s = 5  # Doubled at each retry.
+
+    while True:
+        try:
+            response = urlopen(url)
+            content_length = int(response.headers['content-length'])
```

Comment:
Is this reliable? When I switched the gn/ninja download to come from github I found that the content-length header was not always present.